### ISSUE TITLE
"Avoid Road" doesn't appear to work with online routing engines

### ISSUE THREAD

**Issue by szewachvice** [osmandapp/OsmAnd#17918]
### Routing engine
Issue occurs in online routing engines:

- [ ] OsmAnd's in-app offline routing
- [x] Graphhopper, OSRM

### Routing Profile
Car

### Actual and expected routes

Below is an example of an attempt to route around a way that I would like to avoid. The endpoints are: [1](https://osmand.net/map?pin=34.32771,-116.48414#14/34.32771/-116.48414) and [2](https://osmand.net/map?pin=34.33510,-116.45691#16/34.33510/-116.45691), but the issue occurs regardless of location.

As you can see from the screenshots, Osmand's in-app routing engine respects the "Avoid Road" marker, but fails to incorporate this data point into the online routing engine request to Graphhopper and OSRM.

![Screenshot_20230814-130758~2](https://github.com/osmandapp/OsmAnd/assets/38938265/f2832b66-2e2f-494f-8c37-07717ddfb7a8)

![Screenshot_20230814-130725~2](https://github.com/osmandapp/OsmAnd/assets/38938265/32dc7aa4-a890-45c4-b9b3-11aee9af759a)

![Screenshot_20230814-130700~2](https://github.com/osmandapp/OsmAnd/assets/38938265/9fc6c98b-49f0-4dee-9de7-b17cff678bde)

### Is this a regression?

I have never been able to get this to work in previous versions of Osmand on Android.

## 🌍 Your Environment

**OsmAnd Version:**

<pre><code>

Osmand 4.5.10

</code></pre>

**Device and Android/iOS version:**
Pixel 6, Android 13

**Maps used (online or offline):**
Offiline

- [x] Offline maps offered within the OsmAnd app for download.

- [ ] Online (tile / raster) maps

**Comment by DmitryAlexei** [comment]
Online routing really does not take into account *avoid roads* option (https://osmand.net/docs/user/map/map-context-menu#avoid-road see note in this article). 
The route is fully calculated on third-party servers, OsmAnd settings are not applied (except https://osmand.net/docs/user/navigation/routing/online-routing#online-routing-setting)

**Comment by sonora** [comment]
Actually, several online engines' APIs do have "Avoid" support, so in some cases it may be supportable. But likely tedious.

**Comment by scaidermern** [comment]
Ideally, OsmAnd should only display those settings that are supported by the currently chosen routing engine.

**Comment by szewachvice** [comment]
Exactly: if "Avoid" doesn't work with a given routing engine, it shouldn't be available, otherwise it just feels like a bug.

Also, the Valhalla API ([which Osmand doesn't interface with yet)](https://github.com/osmandapp/OsmAnd/issues/17917) supports off-limits polygons.

**Comment by sonora** [comment]
@scaidermern , @szewachvice Yes, true. Graying out would be preferably over total hiding, though, so that as a user you can directly verify which particlar feature is not available for the selected engine.
